src/test/java/org/jongo/model/IdSpecSet.java

```diff
@@ -0,0 +1,143 @@
+package org.jongo.model;
+
+import java.lang.reflect.Field;
+
+import org.bson.types.ObjectId;
+import org.jongo.marshall.jackson.oid.MongoId;
+import org.jongo.marshall.jackson.oid.MongoObjectId;
+
+import com.fasterxml.jackson.annotation.*;
+
+/**
+ * A set of example id specifications.
+ * 
+ * @author Christian Trimble
+ */
+public class IdSpecSet {
+  public static class BrokenStringIdField {
+      public String id;
+  }
+
+  public static class StringIdBare {
+      public String _id;
+  }
+  
+  public static class StringIdJsonProperty {
+      @JsonProperty("_id")
+      public String id;
+  }
+  
+  public static class BrokenStringIdJsonProperty {
+      @JsonProperty("id")
+      public String id;
+  }
+  
+  public static class StringIdMongoId {
+      @MongoId
+      public String id;
+  }
+
+  public static class StringIdMongoObjectId {
+      @MongoObjectId
+      public String _id;      
+  }
+  
+  public static class StringIdMongoIdMongoObjectId {
+      @MongoId
+      @MongoObjectId
+      public String id;
+  }
+
+  public static class ObjectIdBare {
+    public ObjectId _id;
+  }
+  
+  public static class ObjectIdJsonProperty {
+      @JsonProperty("_id")
+      public ObjectId id;
+  }
+  
+  public static class ObjectIdMongoId {
+      @MongoId
+      public ObjectId id;
+  }
+
+  public static class ObjectIdMongoObjectId {
+      @MongoObjectId
+      public ObjectId _id;      
+  }
+  
+  public static class ObjectIdMongoIdMongoObjectId {
+      @MongoId
+      @MongoObjectId
+      public ObjectId id;
+  }
+  
+  public static abstract class StringIdMongoIdMixIn {
+      @MongoId
+      public String id;
+  }
+  
+  public static abstract class String_IdMongoObjectIdMixIn {
+      @MongoObjectId
+      public String _id;
+  }
+  
+  public static abstract class StringIdMongoObjectIdMixIn {
+      @MongoObjectId
+      public String id;
+  }
+
+  public static abstract class StringIdMongoIdMongoObjectIdMixIn {
+      @MongoId
+      @MongoObjectId
+      public String id;
+  }
+
+  public static <T> T id( T target, Object id ) {
+      Field field = idField(target.getClass());
+      try {
+      field.setAccessible(true);
+      if( ObjectId.class.isAssignableFrom(field.getType()) ) {
+          field.set(target,  (ObjectId)id);
+      }
+      else {
+          field.set(target, id.toString());
+      }
+      return target;
+      } catch( Exception e ) {
+          throw new RuntimeException("could not set id field", e);
+      }
+  }
+  
+  public static Object id(Object target) {
+      Field field = idField(target.getClass());
+
+      field.setAccessible(true);
+      try {
+        return field.get(target);
+    } catch (Exception e) {
+        throw new RuntimeException("could not set id", e);
+    } 
+  }
+  
+  public static <T> T specInstance(Class<T> cls, ObjectId id) {
```

Comment:
Oh, perhaps you need `specInstance`.  I can see that.  Perhaps `newInstanceForSpec` with the class argument named `spec` would clear things up.